src/Aspire.Dashboard/ResourceService/DashboardClient.cs

```diff
@@ -517,9 +537,29 @@ async IAsyncEnumerable<IReadOnlyList<ResourceLogLine>> IDashboardClient.Subscrib
             {
                 yield return batch.SelectMany(batch => batch).ToList();
             }
+
+            if (cancelAfterInitialData)
+            {
+                await initialDataCts.CancelAsync().ConfigureAwait(false);
+                break;
+            }
```

Comment:
It's possible that big logs will be returned in multiple batches.

Reply:
All batches are returned on line #538 - that hasn't changed